### ISSUE TITLE
post/linux/gather/enum_system faults on use

### ISSUE THREAD

**Issue by jmhickman** [rapid7/metasploit-framework#8127]
## Steps to reproduce

How'd you do it?

1. Established mettle x86 reverse TCP session from target  
2. backgrounded session, loaded up `post/linux/gather/enum_system` module, set SESSION var to mettle meterpreter session  
3. Ran module

## Expected behavior  
Output enumerating the target through the active session  

## Current behavior
```
msf post(enum_system) > set SESSION 7
SESSION => 7
msf post(enum_system) > run

[-] Post failed: NoMethodError undefined method `process' for nil:NilClass
[-] Call stack:
[-]   /opt/metasploit/lib/msf/core/post/common.rb:110:in `cmd_exec'
[-]   /opt/metasploit/lib/msf/core/post/linux/system.rb:19:in `get_sysinfo'
[-]   /opt/metasploit/modules/post/linux/gather/enum_system.rb:37:in `run'
[*] Post module execution completed
```

## System stuff  
Target is Ubuntu 12.04, running kernel 3.2.0-23 generic  
Attacking system is running linux 4.10.3-1-ARCH  
### Metasploit version
```
msf exploit(handler) > version
Framework: 4.13.22-dev
Console  : 4.13.22-dev
```  
### I installed Metasploit with:
- [ ] Kali package via apt
- [ ] Omnibus installer (nightly)
- [ ] Commercial/Community installer (from http://www.rapid7.com/products/metasploit/download.jsp)
- [ ] Source install (please specify ruby version)

### OS

Arch linux from ArchStrike repo

**Comment by rwhitcroft** [comment]
I can't reproduce this. Are you sure the session was fully established and working before you ran the post module?

**Comment by jmhickman** [comment]
Yes. I went back to the session after the error and worked with a shell for another 20-30 min before closing the session.

I went ahead and tried it again, and this time instead of the trace I posted above, the payload on the target segfaults and the session dies as soon as I run the post module. Before running the module I can interact with the session normally, or even just leave it idle, and nothing strange happens.

Would a `strace` of the payload when it segfaults be useful? `gdb` isn't on the system and I don't know how to use it.
